cli/command/image/build.go

```diff
@@ -662,13 +662,22 @@ func parseOutputs(inp []string) ([]types.ImageBuildOutput, error) {
 		if err != nil {
 			return nil, err
 		}
-		if len(fields) == 1 && fields[0] == s {
-			outs = append(outs, types.ImageBuildOutput{
-				Type: "local",
-				Attrs: map[string]string{
-					"dest": s,
-				},
-			})
+		if len(fields) == 1 && fields[0] == s && !strings.HasPrefix(s, "type=") {
```

Comment:
I feel this behavior is confusing

Reply:
why?

Reply:
If `docker build -o -` produces tar, user would also except `docker build -o a` to produce tar

Reply:
csv is the main format. non csv is an output path similar to many other tools. And `-` path does not make sense as a directory name, so as many other tools it means outputting to stdout(and we can't write a directory to stdout).

> If docker build -o - produces tar, user would also except docker build -o a to produce tar

Somewhat yes, but docker build always produces a directory(list of files), not a single file. Most common (and promoted) way to use it should be `docker build -o . .` meaning input from current and output to current, while `cat in.tar | docker build -o - - > out.tar` also follows same pattern.

Reply:
Note that we do not support `docker build ctx.tar` either.